pkg/webui/console/containers/gateway-managed-gateway/connection-settings/index.js

```diff
@@ -312,46 +330,52 @@ const GatewayConnectionSettings = () => {
 
       setError(undefined)
       try {
+        const body = {}
         const { wifi_profile, ethernet_profile } = values
 
-        const shouldUpdateNonSharedWifiProfile =
-          wifi_profile.profile_id === 'non-shared' &&
-          Boolean(nonSharedWifiProfileId) &&
-          wifi_profile._enable_wifi_connection
+        if (hasWifi) {
+          const shouldUpdateNonSharedWifiProfile =
+            wifi_profile.profile_id === 'non-shared' &&
+            Boolean(nonSharedWifiProfileId) &&
+            wifi_profile._enable_wifi_connection
 
-        const wifiProfileId = await getWifiProfileId(wifi_profile, shouldUpdateNonSharedWifiProfile)
-        const ethernetProfileId = await getEthernetProfileId(
-          ethernet_profile,
-          cleanValues.ethernet_profile,
-        )
-        const body = {
-          wifi_profile_id:
-            !wifi_profile._enable_wifi_connection && !wifi_profile._override ? null : wifiProfileId,
-          ethernet_profile_id: ethernetProfileId,
+          if (!wifi_profile._enable_wifi_connection && !wifi_profile._override) {
+            body.wifi_profile_id = null
+          } else {
+            body.wifi_profile_id = await getWifiProfileId(
+              wifi_profile,
+              shouldUpdateNonSharedWifiProfile,
+            )
+          }
+        }
+        if (hasEthernet) {
+          body.ethernet_profile_id = await getEthernetProfileId(
+            ethernet_profile,
+            cleanValues.ethernet_profile,
+          )
         }
 
         await dispatch(attachPromise(updateManagedGateway(gtwId, body)))
 
         // Reset the form and the initial values
-        let resetValues = { ...values }
-        if (wifi_profile.profile_id !== 'non-shared') {
-          resetValues = {
-            ...values,
-            wifi_profile: {
-              ...values.wifi_profile,
-              ...initialWifiProfile,
-              profile_id: wifiProfileId,
-              _profile_of: wifi_profile._profile_of,
-            },
-          }
+        const resetValues = {
+          ...values,
+          ...(hasWifi &&
+            wifi_profile.profile_id !== 'non-shared' && {
+              wifi_profile: {
+                ...initialWifiProfile,
```

Comment:
One thing I missed during the last review.

```suggestion
                ...wifi_profile,
                ...initialWifiProfile,
```